tests/instances.tcl

```diff
@@ -47,8 +48,9 @@ proc exec_instance {type cfgfile} {
         error "Unknown instance type."
     }
 
+    set errfile [file join $dirname err.txt]
     if {$::valgrind} {
-        set pid [exec valgrind --track-origins=yes --suppressions=../../../src/valgrind.sup --show-reachable=no --show-possibly-lost=no --leak-check=full ../../../src/${prgname} $cfgfile &]
+        set pid [exec valgrind --track-origins=yes --suppressions=../../../src/valgrind.sup --show-reachable=no --show-possibly-lost=no --leak-check=full ../../../src/${prgname} $cfgfile 2>> $errfile &]
```

Comment:
@oranagra Is there a particular reason you're redirecting `stderr` rather than use the valgrind `--log-file` option?

It may not matter if we don't expect `$prgname` to use `stderr` but that may not always be the case and error messages that were printed to console in the past would no longer be visible at real time.

Reply:
@yossigo I followed the same technique that is used by the main redis tests. 
The main redis tests redirect stdout to one file and stderr to another. 
The cluster tests are using the `logfile` config instead of redirecting the stdout. 

Anyway, redis doesn't normally outputs anything to stderr, and if we want to change the valgrind log approach, it's more important to change the main redis test suite too. 
Although come to think of it, both run multiple servers in parallel, so putting things, to stderr rather than a file may be confusing. 

Let me know if you still think this should be changed and if you wanna change both tests infra or just the one